code/modules/surgery/organs/lungs.dm

```diff
@@ -102,6 +102,10 @@
 			H.throw_alert("not_enough_nitro", /atom/movable/screen/alert/not_enough_nitro)
 		return FALSE
 
+	var/filtering_gas_mask = FALSE
+	if(H.wear_mask && H.wear_mask.clothing_flags & GAS_FILTERING && H.wear_mask.consume_filter())
+		filtering_gas_mask = TRUE
```

Comment:
I admit, I don't like how this is handled.

It'd be a lot more interesting if the gas actually modified the the `gas_mixture`, and either stored the gas removed in the filter, or something like that---you could have it so the amount of nasty gas breathed impacted how quickly the gas mask's filter decays---so being in a very very high concentrations of nasty stuff™ wears out the filter faster than if you're just in an atmosphere that's 0.001% nasty stuff™.